File: packages/javascript/lib/word/randDrink.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Something to drink, hot, cold or fermented.
 *
 * `randWord({ theme: 'drink' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randDrink({ language: 'ko', count: 3 }); // ['식혜', '보리차', '막걸리']
 * randDrink({ language: 'en', count: 3 }); // ['Cider', 'Cordial', 'Lemonade']
 */
export function randDrink(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Drink words with the language and theme each one came from.
 *
 * @example
 * randDrink({ language: 'ko', output: 'detail' });
 * // [{ word: '식혜', language: 'ko', theme: 'drink' }]
 */
export function randDrink(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randDrink(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('drink', options);
}

File: packages/dart/lib/src/word/rand_drink.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Something to drink, hot, cold or fermented.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.drink] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randDrink(language: WordLanguage.ko, count: 3); // [식혜, 보리차, 막걸리]
/// randDrink(language: WordLanguage.en, count: 3); // [Cider, Cordial, Lemonade]
/// ```
List<String> randDrink({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.drink,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
